### PR TITLE
back button takes to main screen fix

### PR DESCRIPTION
**Pull request by chhavip**
Made changes in Completed Scene class to make the back button take us back to the map instead of going through all the previous screens again.

### CODE REVIEW

**Comment by chhavip** [comment]
Have changed the behavior by overriding the OnBackPressed in CompletedScene. This is a squashed commit for both

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public void onClick(View v) {
 			} else {
 				Intent myIntent = new Intent(MapActivity.this, CompletedScene.class);
 				startActivityForResult(myIntent, 0);
+				finish();
```
Comment (alma-castant) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: What is the goal of this finish call here?

Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: If it is not called, this map activity remains active which causes two map activities to run in the background when we try to come back from completed scene, I tried it and it was leading to reappearance of map activity

File: PowerUp/app/src/main/java/powerup/systers/com/CompletedScene.java [hunk 2]
```diff
@@ -20,8 +20,16 @@ public void onClick(View v) {
...
 				startActivityForResult(myIntent, 0);
+				finish();
 			}
 		});
 	}
-	
+
+	@Override
+	public void onBackPressed() {
+		super.onBackPressed();
```
Comment (alma-castant) [inline PowerUp/app/src/main/java/powerup/systers/com/CompletedScene.java]: You shouldn't be calling `super.onBackPressed()`. You don't want it to act the way it did before! :)

File: PowerUp/app/src/main/java/powerup/systers/com/CompletedScene.java [hunk 3]
```diff
@@ -20,8 +20,16 @@ public void onClick(View v) {
...
+		Intent i = new Intent(CompletedScene.this,MapActivity.class);
```
Comment (alma-castant) [inline PowerUp/app/src/main/java/powerup/systers/com/CompletedScene.java]: Create a better name for the Intent. Something like `intent`. You could also declare it in line when you call `startActivity`

**Comment by alma-castant** [comment]
I added two comments about the code. You should also be testing the feature actually works the way it should before creating the PR. Currently when you press back on the game over activity, it goes to the previous conversation step. You should add a `finish()` call after you go to the game over activity from the game activity

File: PowerUp/app/src/main/java/powerup/systers/com/CompletedScene.java (hunk 3 above)
Comment (alma-castant) [inline PowerUp/app/src/main/java/powerup/systers/com/CompletedScene.java]: Also, please add a space between the two parameters when you call the constructor

**Comment by chhavip** [comment]
I will start this task again from scratch soon taking care of all the corner cases.

**Pull request closed without merging**